src/TrackGenerator.cpp

```diff
@@ -324,23 +355,30 @@ FP_PRECISION TrackGenerator::getFSRVolume(int fsr_id) {
  */
 FP_PRECISION TrackGenerator::getMaxOpticalLength() {
 
-  segment* curr_segment;
-  FP_PRECISION length;
-  Material* material;
-  FP_PRECISION* sigma_t;
   FP_PRECISION max_optical_length = 0.;
 
-  /* Iterate over all tracks, segments, groups to find max optical length */
-  for (int i=0; i < _num_azim; i++) {
-    for (int j=0; j < _num_tracks[i]; j++) {
-      for (int s=0; s < _tracks[i][j].getNumSegments(); s++) {
-        curr_segment = _tracks[i][j].getSegment(s);
-        length = curr_segment->_length;
-        material = curr_segment->_material;
-        sigma_t = material->getSigmaT();
+#pragma omp parallel
+  {
+
+    segment* curr_segment;
+    FP_PRECISION length;
+    Material* material;
+    FP_PRECISION* sigma_t;
 
-        for (int e=0; e < material->getNumEnergyGroups(); e++)
-          max_optical_length = std::max(max_optical_length, length*sigma_t[e]);
+    /* Iterate over all tracks, segments, groups to find max optical length */
+    for (int i=0; i < _num_azim; i++) {
+#pragma omp for reduction(max:max_optical_length)
```

Comment:
The max optical length is first computed here, but the parallel region in `TrackGenerator::splitSegments()` where segments are cut up based on their length w.r.t. the max optical path length is where the `TAU_NUDGE` manifested itself. This parallel region *might* bear some responsibility as well though.